File: rrhh-service/src/main/java/pe/albrugroup/rrhh_service/entity/request/RegistrarContratoRequest.java
package pe.albrugroup.rrhh_service.entity.request;

import jakarta.validation.constraints.*;
import lombok.*;
import pe.albrugroup.rrhh_service.entity.enums.*;

import java.math.BigDecimal;
import java.time.LocalDate;

@Builder(toBuilder = true) @Getter @Setter
@AllArgsConstructor @NoArgsConstructor
public class RegistrarContratoRequest {

    // PUESTO DE TRABAJO
    @NotNull private PuestoTrabajo puestoTrabajo;
    @NotNull private Regimen regimen;
    @NotNull private Modalidad modalidad;
    @NotNull private SeguroSalud seguroSalud;
    @NotNull private SistemaPensiones sistemaPensiones;
    @NotNull @DecimalMin(value = "0.00", inclusive = false)
    @Digits(integer = 10, fraction = 2)
    private BigDecimal sueldoBase;
    // VIGENCIA
    @NotNull private LocalDate fechaInicio;
    private LocalDate fechaFin;
}

File: frontend/src/app/shared/models/rrhh/registrar-contrato-request.ts
export interface RegistrarContratoRequest {
  idPostulacion?: number | null;
  categoriaPersonal?: 'ESTRUCTURAL' | 'OPERATIVO';
  puestoTrabajo?: string | null;
  regimen: string;
  modalidad: string;
  seguroSalud?: string | null;
  sistemaPensiones?: string | null;
  sueldoBase: number;
  fechaInicio: string;
  fechaFin?: string | null;
}
